sdks/python/apache_beam/runners/interactive/display/pcoll_visualization.py

```diff
@@ -0,0 +1,279 @@
+            <script src="https://ajax.googleapis.com/ajax/libs/jquery/2.2.2/jquery.min.js"></script> 
+            <script src="https://cdn.datatables.net/1.10.16/js/jquery.dataTables.js"></script> 
+            <link rel="stylesheet" href="https://cdn.datatables.net/1.10.16/css/jquery.dataTables.css">
+            {dataframe_html}
+            <script>
+              $("#{table_id}").DataTable();
+            </script>"""
+
+
+def visualize(pcoll, dynamic_plotting_interval=None):
+  """Visualizes the data of a given PCollection. Optionally enables dynamic
+  plotting with interval in seconds if the PCollection is being produced by a
+  running pipeline or the pipeline is streaming indefinitely. The function
+  always returns immediately and is asynchronous when dynamic plotting is on.
+
+  If dynamic plotting enabled, the visualization is updated continuously until
+  the pipeline producing the PCollection is in an end state. The visualization
+  would be anchored to the notebook cell output area. The function
+  asynchronously returns a handle to the visualization job immediately. The user
+  could manually do::
+
+    # In one notebook cell, enable dynamic plotting every 1 second:
+    handle = visualize(pcoll, dynamic_plotting_interval=1)
+    # Visualization anchored to the cell's output area.
+    # In a different cell:
+    handle.stop()
+    # Will stop the dynamic plotting of the above visualization manually.
+    # Otherwise, dynamic plotting ends when pipeline is not running anymore.
+
+  If dynamic_plotting is not enabled (by default), None is returned.
+  """
+  if not _ipython_ready:
+    return None
+  pv = PCollVisualization(pcoll)
+  pv.display_facets()
+
+  if dynamic_plotting_interval and _tl_ready:
+    # Disables the verbose logging from timeloop.
+    logging.getLogger('timeloop').disabled = True
+    tl = Timeloop()
+
+    def dynamic_plotting(pcoll, pv, tl):
+      @tl.job(interval=timedelta(seconds=dynamic_plotting_interval))
+      def continuous_update_display():  # pylint: disable=unused-variable
+        # Always creates a new PCollVisualization instance when the
+        # PCollection materialization is being updated and dynamic
+        # plotting is in-process.
+        updated_pv = PCollVisualization(pcoll)
+        updated_pv.display_facets(updating_pv=pv)
+        if ie.current_env().is_terminated(pcoll.pipeline):
+          try:
+            tl.stop()
+          except RuntimeError:
+            # The job can only be stopped once. Ignore excessive stops.
+            pass
+
+      tl.start()
+      return tl
+
+    return dynamic_plotting(pcoll, pv, tl)
+  return None
+
+
+class PCollVisualization(object):
+  """A visualization of a PCollection.
+
+  The class relies on creating a PipelineInstrument w/o actual instrument to
+  access current interactive environment for materialized PCollection data at
+  the moment of self instantiation through cache. It utilizes Facets, pandas
```

Comment:
I think you can drop the last sentence. What it uses to visualize does not need to be part of the documentation.